example/simple_testing/kotlin/src/main/kotlin/dev/diplomattest/simple_testing/Main.kt

```diff
@@ -0,0 +1,128 @@
+    @JvmField
+    internal var vtable: DiplomatTrait_TesterTrait_VTable_Native
+        = DiplomatTrait_TesterTrait_VTable_Native();
+  
+    // Define the fields of the struct
+    override fun getFieldOrder(): List<String> {
+        return listOf("data_", "vtable")
+    }
+}
+
+class DiplomatCallback_TesterTrait_Wrapper internal constructor (
+    internal val nativeStruct: DiplomatTrait_TesterTrait_Wrapper_Native) {
+    val data_: Pointer = nativeStruct.data_
+    val vtable: DiplomatTrait_TesterTrait_VTable_Native = nativeStruct.vtable
+
+    companion object {
+        val NATIVESIZE: Long = Native.getNativeSize(DiplomatTrait_TesterTrait_Wrapper_Native::class.java).toLong()
+        internal val libClass: Class<DiplomatCallback_Lib> = DiplomatCallback_Lib::class.java
+        internal val lib: DiplomatCallback_Lib = Native.load("somelib", libClass)
+
+        fun fromTraitObj(trt_obj: DiplomatTraitInterface_TesterTrait): DiplomatCallback_TesterTrait_Wrapper {
+            val test_trait_fn: DiplomatCallback_Lib.Runner_DiplomatTraitMethod_TesterTrait_test_trait_fn = object :  DiplomatCallback_Lib.Runner_DiplomatTraitMethod_TesterTrait_test_trait_fn {
+                override fun invoke(ignored: Pointer?, input: Int): Int {
+                    return trt_obj.test_trait_fn(input);
+                }
+            }
+            val test_void_trait_fn: DiplomatCallback_Lib.Runner_DiplomatTraitMethod_TesterTrait_test_void_trait_fn = object :  DiplomatCallback_Lib.Runner_DiplomatTraitMethod_TesterTrait_test_void_trait_fn {
+                override fun invoke(ignored: Pointer?): Unit {
+                    trt_obj.test_void_trait_fn();
+                }
+            }
+            val vtable = DiplomatTrait_TesterTrait_VTable_Native()
+            vtable.run_test_trait_fn_callback = test_trait_fn;
+            vtable.run_test_void_trait_fn_callback = test_void_trait_fn;
+            val native_wrapper = DiplomatTrait_TesterTrait_Wrapper_Native();
+            native_wrapper.vtable = vtable;
+            return DiplomatCallback_TesterTrait_Wrapper(native_wrapper);
+        }
+
+        fun test_trait_fn(trait_wrap: DiplomatCallback_TesterTrait_Wrapper, i: Int): Int {
+            return lib.Wrapper_test_with_trait(trait_wrap.nativeStruct, i);
+        }
+    }
+
+}
+
+interface DiplomatCallback_Lib: Library {
+    interface Runner_DiplomatTraitMethod_TesterTrait_test_trait_fn: Callback {
+        fun invoke(ignored: Pointer?, input: Int): Int
+    }
+
+    interface Runner_DiplomatTraitMethod_TesterTrait_test_void_trait_fn: Callback {
+        fun invoke(ignored: Pointer?): Unit
+    }
+
+    fun Wrapper_test_with_trait(diplomatTraitWrapper: DiplomatTrait_TesterTrait_Wrapper_Native, i: Int): Int
+}
+
+// the Kotlin interface that you'll extend to have access to the Rust trait `TesterTrait`
+interface DiplomatTraitInterface_TesterTrait {
+    fun test_trait_fn(x: Int): Int;
+    fun test_void_trait_fn(): Unit;
+}
```

Comment:
If I were a consumer of this API I'd prefer to use less boilerplate in names and style conforming to the host language. For the internal stuff it doesn't matte but is there any reason we couldn't do this?

```suggestion
interface TesterTrait {
    fun testTraitFn(x: Int): Int;
    fun testVoidTrait_fn(): Unit;
}
```
It feels like it be much more ergonomic to the person whose using the library on the kotlin side.

Reply:
Great point! we'll do that.